File: C++/Advanced/Stacks And Queues/Exercise/Fashion Boutique.cpp
#include <iostream>
#include <cstdlib>
#include <stack>
#include <string>

using namespace std;

void loadClothes(stack<int>& clothes, string clothesStr) {
	string numRep = "";
	for (int k = 0; k <= clothesStr.length(); k++) {
		if (clothesStr[k] == ' ' || k == clothesStr.length()) {
			clothes.push(stoi(numRep));
			numRep = "";
			continue;
		}
		else {
			numRep += clothesStr[k];
		}
	}
}

void HangClothes(stack<int> clothes, int rackCapacity, int& usedRacks) {
	int sumCap = 0;
	if (rackCapacity > 0) {
		while (!clothes.empty()) {
			if (sumCap + clothes.top() == rackCapacity) {
				sumCap = 0;
				clothes.pop();
				if (clothes.size() > 0) {
					usedRacks += 1;
				}
				continue;
			}
			else if (sumCap + clothes.top() > rackCapacity) {
				usedRacks += 1;
				sumCap = 0;
				continue;
			}
			else {
				sumCap += clothes.top();
			}
			clothes.pop();
		}
	}
	else {
		usedRacks -= 1;
	}
}

int main() {
	stack<int> clothes;
	string clothesInt;
	int rackCap, usedRacks = 1;
	getline(cin >> ws, clothesInt);
	cin >> rackCap;
	loadClothes(clothes, clothesInt);
	HangClothes(clothes, rackCap, usedRacks);
	cout << usedRacks;
	return 0;
}

File: Python/Advanced/Stacks And Queues/Exercise/Fashion Boutique.py
class Stack():
    stackVar=[]
    def peek(self):
        return self.stackVar[-1]
    def push(self, val):
        self.stackVar.append(val)
    def pop(self):
        self.stackVar.pop()
    def count(self):
        return len(self.stackVar)
    def __repr__(self):
        return "[STACK](["+', '.join(self.stackVar)+"])"

def HangClothes(stack, racksUsed, rackCap):
    sumRackUsage=0
    while stack.count():
        val=stack.peek()
        if val+sumRackUsage>rackCap:
            racksUsed+=1; sumRackUsage=0
            sumRackUsage+=val
        else:
            sumRackUsage += val
        stack.pop()
    return racksUsed


clothesDelivery=input(); rackCap=int(input())
racks=1
clothes=Stack()
clothesDelivery=clothesDelivery.split(" "); clothesDelivery=[int(e) for e in clothesDelivery]

for k in clothesDelivery:
    clothes.push(k)

print(HangClothes(clothes, racks, rackCap))